### PR TITLE
add warning to remind Optimizer() deprecates

### PR DESCRIPTION
**Pull request by Le-Zheng**
## What changes were proposed in this pull request?

Add a deprecate warning

## How was this patch tested?

Jenkins

## Related links or issues (optional)
fix https://github.com/intel-analytics/BigDL/issues/3041 and https://github.com/intel-analytics/BigDL/issues/3040

### CODE REVIEW

File: pyspark/bigdl/optim/optimizer.py [hunk 1]
```diff
@@ -869,6 +871,7 @@ def __init__(self,
                                       bigdl_type)
         self.value = self.pvalue.value
         self.bigdl_type = self.pvalue.bigdl_type
+        assert isinstance(training_rdd, RDD), "Only type of RDD is allowed"
```
Comment (hkvision) [inline pyspark/bigdl/optim/optimizer.py:874]: put this before creating DistriOptimizer?

File: pyspark/bigdl/optim/optimizer.py [hunk 2]
```diff
@@ -860,6 +861,7 @@ def __init__(self,
         :param end_trigger: when to end the optimization
         :param batch_size: training batch size
         """
+        warnings.warn("This is a deprecated method, you should use `create` method instead.")
```
Comment (hkvision) [inline pyspark/bigdl/optim/optimizer.py:864]: It is strange to say __init__ is deprecated?
What about changing to `You are recommended to use create....`

Comment (Le-Zheng) [reply pyspark/bigdl/optim/optimizer.py]: Nice

**Review by hkvision — APPROVED** [review]

**Comment by Le-Zheng** [comment]
http://10.239.47.210:18080/view/PR/job/BigDL-PRVN-lite/1170/ tests passed

**Pull request merged**